---
title: "ns_123"
author: "Bjarkemp"
format: html
      code-fold: true
editor: visual
---

# Basic Population Genetics (NS 1, 2, 3)

## 1. Allele Frequency, Genotype frequency and Hardy Weinberg Equilibrium

### 1.1

A researcher examines a locus in which there is a particular C/T polymorphism. She obtains the following genotypic count:

```{r}
cc <- 42
ct <- 16
tt <- 32
```

**Calculate the genotype frequencies and allele frequencies in the sample:**

**Genotype frequencies**:

```{r}
#your solution here
```

**Solution:**

```{r}
total <- cc+tt+ct
cc_freq <- cc/total
ct_freq <- ct/total
tt_freq <- tt/total
```

**Allele frequencies:**

```{r}
#your solution here
```

**Solution:**

```{r}
c_freq <- (2*cc+ct)/(2*total)
t_freq <- (2*tt+ct)/(2*total)
c_freq+t_freq
```

### 1.2

**For the data from exercise 1.1, find the expected homozygosity and the expected heterozygosity, given the observed allele frequencies, and calculate the inbreeding coefficient (F).**

The inbreeding coefficient is just another way to measure the decrease in heterozygosity in a population beyond that expected under HWE.

```{r}

```

**Solution:**

```{r}
e_homo <- c_freq**2+t_freq**2
e_hetero <- 2*c_freq*t_freq
e_homo+e_hetero
inbreed_f <- (e_hetero-ct_freq)/e_hetero
```

if F=0 the population is in HWE and F=1 we have no heterozygotes in the population.

### 1.3

**For the data in Exercise 1.1, test if the population is in HWE using A chi-square test at the 5% significance level**

```{r}

```

**Solution:**

```{r}
df <- tibble(state=c('CC','CT','TT'),
             observed=c(cc,ct,tt),
             expected=c(total*(c_freq**2),
                        total*2*c_freq*t_freq,
                        total*(t_freq**2)))

chisq.test(x = df$observed, y = df$expected) #lazy i know
```

### 1.4

**The proportion of a population suffering from a specific rare genetic disease is 0.02%. assume that the disease is caused by a single recessive allele and assume that the population is in HWE. How many individuals carry the disease allele in the heterozygous state?**

```{r}

```

**Solution:**

```{r}
#We know that 0.02% suffers from the disease, and sice we expect HWE, we can
# assume that this must be the genotype frequency, the genotype frequency is fa**2 
# the allel freq is then sqrt(faa) 
# as the system is biallelic we know that the freq of the healthy allel is then 1-fa 
a_freq <- sqrt(0.02)
A_freq <- (1-a_freq)
#the expected frequency of carriers is then 2fafA
2*a_freq*A_freq
```

### 1.5

In another locus there are three alleles, A, C , T and the genotypic counts in the sample are

```{r}
AA <- 10
AC <- 10
AT <- 5
CC <- 20
CT <- 5
TT <- 20 
```

```{r}

```

**Solution:**

```{r}
total <- AA+AC+AT+CC+CT+TT
A_freq <- (2*AA+AC+AT)/total
T_freq <- (2*TT+CT+AT)/total
C_freq <- (2*CC+CT*AC)/total
```

### 1.6

**For the data from Exercise 1.5, find the expected homozygosity and the expected heterozygosity, given the observed allele frequencies.**

```{r}

```

**Solution:**

```{r}
e_homo <- (A_freq**2)+(C_freq**2)+(T_freq**2)
e_hetero <- 2*A_freq*C_freq+2*A_freq*T_freq+2*C_freq*T_freq
```

### 1.7

**For the data in Exercise 1.5, test if the population is in HWE using a chi-square test at the 5% significance level.**

```{r}

```

**Solution:**

```{r}
df <- tibble(
  state = c('AA','TT','CC','AT','AC','CT'),
  observed = c(AA, TT, CC, AT, AC, CT),
  expected = c((A_freq**2)*total, (T_freq**2)*total,
               (C_freq**2)*total, 2*A_freq*T_freq,
               2*A_freq*C_freq,2*C_freq*T_freq))
chisq.test(df$observed,df$expected)
```

### 1.8

An individual has genotype CT for the locus discussed in Exercise 1.1 and genotype AC discussed in 1.5. At a crime scene, forensic evidence is found with the exact same genotype (CT, AC). What is the chance of such a match by random assuming HWE and the allele frequencies in 1.1 and 1.5?

```{r}

```

**Solution:**

```{r}
#1.1
cc <- 42
ct <- 16
tt <- 32
total <- cc+ct+tt
# observed
c_freq <- (2*cc+ct)/(2*total)
t_freq <- (2*tt+ct)/(2*total)
#expected
exp_CT <- 2*c_freq*t_freq

#1.5
AA <- 10
AC <- 10
AT <- 5
CC <- 20
CT <- 5
TT <- 20 
total <- AA+AC+AT+CC+CT+TT
#observed
A_freq <- (2*AA+AC+AT)/total
C_freq <- (2*CC+CT*AC)/total
#expected AC
exp_AC <- 2*A_freq*C_freq
# If we assume that these two loci are sufficeintly far a part and thus independent the #probability of having both is simply, the chance of having one times the chance of having the # other 
(prob <- exp_AC*exp_CT)

```

What is the match probability if the calculations is done using observed genotypes instead?

```{r}

```

**Solution:**

```{r}
#observed: 
ct_freq <- ct/total
ac_freq <- AC/total
(prob <- ct_freq*ac_freq)
```

### 1.9

Show mathematically that it takes two generations to achieve HWE when allele frequencies differ between males and females

Here I find it easier to do on paper, so you might benefit by doing that too

**Solution:**

```{r}
# Ha you will have to look elsewhere to find this ;)
```

# 2. Genetic drift and Mutation

### 2.1

At a particular locus, there are four nucleotides segregating: A, C, T, G with frequencies in the population of 0.1, 0.2, 0.2 and 0.5, respectively. Assume that the population evolves in accordance with a wright-fisher model, with no new mutations. What is the probability that nucleotide A at this locus will eventually be lost from the population? What is the probability that either A or C will go to fixation?

```{r}

```

**Solution:**

```{r}
#A will be lost
1-0.1
# A or C fix
0.1+0.2
```

### 2.2

An insertion mutation is segregating in a population at a frequency of 10%. Each generation, in each individual carrying the mutation, the probability that this mutatutaion is reverted to the ancestral form by a deletion is 10\*\*-6. The mutation occurs de novo, in individuals not carrying it at a rate og 5\*10\*\*-6 per generation if there is no genetic drift or selection, what is the expected frequency of the mutation after one generation?

```{r}

```

**Solution:**

```{r}
(1-(10**-6))*0.1+(5*10**-6)*0.9
```

### 2.3

What is the expected equilibrium frequency of the mutation discussed in 2.2?

```{r}

```

**Solution:**

```{r}
(5*10**-6)/((5*10**-6)+(10**-6))
```

### 2.4

The mutation rate in a particular gene is 1\*10\*\*-9 per generation per base pair. The gene is 800bp long. Assume that both humans and chimpanzees have a generation time of twenty years and that each mutation will create a new nucleotide difference between chimps and humans. If the divergence time is 6 million years how many differences in this gene would you expect to observe between humans and chimps.

```{r}

```

**Solution:**

```{r}
t <- (6*10**6)/20
mu <- 1*10**-9
mut_in_gene <- mu * 800 

expected_diff <- 2*t*mut_in_gene
```

### 2.5

The divergence time between two species of fish (A and B) is 20\*10\*\*6 years. In a particular gene, they differ by 29 nucleotide differences. The number of nucleotide differences between A and another species C is 12. Based on this information and the assumption of a molecular clock with a constant generation time, provide an estimate of the divergence between A and C.

```{r}

```

**Solution**

```{r}
per_year <- 29/(2*20*10**6)
(a_2_c <- 12/(2*per_year))
```

### 2.6

Consider the standard neutral Wright-Fisher model with population size 2N. What is the expected number of offspring of a particular individual in the next generation:

### 2.7

For the model in Exercise 2.6, write a formula for the probability that a particular individual from generation t leaves no descendants in generation t + 1.

### 2.8

For the model in Exercise 2.6, write a mathematical formula for the probability that a mutation of frequency p is lost from the population within one generation.

### 2.9

For the model in Exercise 2.6, what is the probability that two individuals in generation t + 1 both have the same parent in the previous generation?

## Coalescence Theory: Relating Theory to Data

### 3.1

In a sample of five gene copies, what is the expected time to the most recent common ancestor under the standard coalescence model measured in 2N generations? What is the expected total tree length?

```{r}

```

**Solution:**

```{r}
#tMRCA
2-2/5 
# or the more conventional
2/(2*(2-1))+2/(3*(3-1))+2/(4*(4-1))+2/(5*(5-1))
#total tree length
2*((1/1)+(1/2)+(1/3)+(1/4)
```

### 3.2

A researcher sequences a 10-kb-long DNA sequence from a single individual. The mutation rate in the region is 10-9 per site. The researcher finds that 21 of the sites are heterozygous. Assuming an infinite sites model and a standard neutral coalescence model, provide an estimate of the effective population size of the population from which this individual has been sampled.

```{r}

```

**Solution:**

```{r}
diff <- 21/(10**4)
(diff/(10**-9))/2 #2N
# N
(diff/(10**-9))/4
```

### 3.3

A researcher sequences 5 diploid individuals (10 DNA sequences)from a population with an effective population size of 20,000 individuals. The total mutation rate for the region is 10-5 per generation. Assuming a standard coalescence model and infinite sites mutation, how many segregating sites should the researcher expect to find in the data?

```{r}

```

**Solution:**

```{r}
4*(2*10**4)*(10**-5)*2.83
#2.83 is sum(1/k from 1 to 9) 
```

### 3.4

For the data and assumptions in Exercise 3.3, what is the expected haplotype homozygosity?

```{r}

```

**Solution:**

```{r}
1-(4*20000*10**-5)/(1+4*20000*10**-5)
```

### 3.5

The following DNA sequence data were obtained from a single population:

AAGATGACAGATAGGCA CTGGTGACTGATAGGCA CTGGTGACTGATAGGCT CAGATGACTGATAGGCT

How many segregating; sites are there? What is the average number of pairwise differences (pi)?

```{r}

```

**Solution:**

```{r}
# 5 segregating sites
#pairwise diffs
(3*1+2*2+2*2+3*1+2*2)/choose(4,2)

```

### 3.6

Using the data from exercise 3.5, calculate two estimates of 0, one based on Watterson's estimator and one based on Tajima's estimator.

```{r}

```

**Solution:**

```{r}
#tajimas
3
#watterson
5/(1/1+1/2+1/3)
```

### 3.7

Make a histogram of the site frequency spectrum (SFS) for the data in Exercise 3.5.

```{r}

```

**Solution:**

```{r}
tibble(shared = c('1','2'), counts = c(2,3)) %>% 
  ggplot(aes(x = shared, y = counts))+
  geom_histogram(stat = 'identity')
```

### 3.8

Do the data in Exercise 3.5 contain more, fewer, or the same number of singletons as expected under the standard neutral coalescence model?

```{r}

```

**Solution:**

```{r}
(1+1/3)/(1+1/2+1/3)
```

### 3.9

A researcher obtains a sample of two diploid individuals (four gene copies) from a population for which the assumptions of the standard coalescence model holds true. What is the probability that the two sequences from the first individual are more closely related to each other than they are to any of the sequences from the second individual, i.e., that they share a most recent common ancestor more recently with each other than with any of the other two sequences?
